src/data/datasets/detection.py

```diff
@@ -0,0 +1,136 @@
+    :param min_area: Value in pixels. If the area of a bounding box after 
+     augmentation becomes smaller than min_area, Albumentations will drop that box
+    :param min_visibility: Value between 0 and 1. If the ratio of the bounding box area after augmentation 
+     to the area of the bounding box before augmentation becomes smaller than min_visibility, 
+     Albumentations will drop that box.
+    """
+    def __init__(self, 
+                    target_column: str = 'annotation',
+                    min_area: float = 0.0,
+                    min_visibility: float = 0.0,
+                    **dataset_params
+    ):
+        super().__init__(**dataset_params)
+        
+        self.target_column = target_column
+        if self.augment is not None:
+            self.augment = module_transforms.Compose(
+                self.augment,
+                bbox_params=module_transforms.BboxParams(
+                    format='pascal_voc',
+                    label_fields=['category_ids'],
+                    min_area=min_area,
+                    min_visibility=min_visibility
+                    )
+            )
+
+        self.transform = module_transforms.Compose(
+                self.transform,
+                bbox_params=module_transforms.BboxParams(
+                    format='pascal_voc',
+                    label_fields=['category_ids'],
+                    min_area=min_area,
+                    min_visibility=min_visibility
+                    )
+            )
+
+        self.csv[target_column] = self.csv[target_column].apply(eval)
+
+    def __getitem__(self, idx: int):
+        sample = self.get_raw(idx // self.expand_rate)
+        sample['image'] = sample['image'].type(torch.__dict__[self.input_dtype])
+        
+        output = {
+            'input': sample['image'],
+            'target_bboxes': torch.tensor(sample['bboxes']).type(torch.__dict__[self.target_dtype]),
+            'target_labels': torch.tensor(sample['category_ids']).type(torch.__dict__[self.target_dtype]),
+            'bbox_count': torch.tensor(sample['bbox_count'])
+        }
+
+        return output
+        
+    def get_raw(self, idx: int):
+        record = self.csv.iloc[idx]
+        image = self.read_image(record)
+        row_annotations = record[self.target_column]
+
+        bboxes = []
+        labels = []
+        for annotation in row_annotations:
+            bbox = [annotation['x_min'], annotation['y_min'], annotation['x_max'], annotation['y_max']]
+            label = annotation['label']
+            bboxes.append(bbox)
+            labels.append(label)
+
+        bbox_count = len(bboxes)
+
+        sample = {
+            'image': image,
+            'bboxes': bboxes,
+            'category_ids': labels
+            }
+
+        if self.augment is not None:
+            sample = self.augment(**sample)
+
+        sample = self.transform(**sample)
+        sample['bbox_count'] = bbox_count
```

Comment:
Since Albumentations can remove some bbox, the real number of bboxes can change. You should calc count after augmentations are applied